### DISCUSSION TITLE
More doc on overriding helptext

### DISCUSSION THREAD

**Discussion by wayneyaoo** [bilal-fazlani/commanddotnet#481] [Q&A] [answered]
Currently the option and its help text are printed vertically like:
```
Options:

  -v | --Verbose
  Increase verbosity
```

I'd like to make it side by side like most app does

```
Options:

  -v | --Verbose                   Increase verbosity
```

I'm looking at the doc here but it doesn't say anything about altering the default template (if this is they way) or create a new template. Do I need to implement a whole new `IHelpProvider` or sub-class the existing one? Any guidance on this is appreciated! Thank you!

**Comment by drewburlingame** [comment, marked as answer]
This is pretty simple to do.  It's touched on briefly here https://commanddotnet.bilal-fazlani.com/help/help/#custom-help-provider. I should add a reference to [BasicHelpTextProvider](https://github.com/bilal-fazlani/commanddotnet/blob/master/CommandDotNet/Help/BasicHelpTextProvider.cs) which serves as an example of subclassing HelpTextProvider.  It's probably close to what you want.  You could use that as an example.

**Reply by wayneyaoo** [reply]
Thanks.. Didn't expect that particular internal class has what I need. But from the implementation it only surfaces argument description. I need other stuff like default values and allowed values. It's a good pointer anyway. I'll dig into the `HelpTextProvider` to see how I can override down the path. Thank you!

**Reply by drewburlingame** [reply]
I think it does have what you need.

There's a private class called ArgumentHelpValues that is [assigned values from virtual methods](https://github.com/bilal-fazlani/commanddotnet/blob/master/CommandDotNet/Help/HelpTextProvider.cs#L296).  Those virtual methods give you what you need.  You can use them in your override of [SectionArguments](https://github.com/bilal-fazlani/commanddotnet/blob/master/CommandDotNet/Help/HelpTextProvider.cs#L123).
